tests/python/pants_test/tasks/test_changed_target_integration.py

```diff
@@ -10,70 +10,91 @@
 
 from pants.base.build_environment import get_buildroot
 from pants.util.contextutil import temporary_dir
-from pants.util.dirutil import safe_open
+from pants.util.dirutil import fast_relpath, safe_file_dump, safe_mkdir, safe_open
 from pants_test.pants_run_integration_test import PantsRunIntegrationTest
 from pants_test.testutils.git_util import initialize_repo
 
 
 class ChangedTargetGoalsIntegrationTest(PantsRunIntegrationTest):
   @contextmanager
   def known_commits(self):
+    """Creates an anonymous git repository under the buildroot."""
     with temporary_dir(root_dir=get_buildroot()) as worktree:
-      with safe_open(os.path.join(worktree, 'README'), 'w') as fp:
-        fp.write('Just a test tree.')
+      rel_buildroot = fast_relpath(worktree, get_buildroot())
+      with self.known_commits_in(worktree):
+        yield worktree
 
-      with initialize_repo(worktree=worktree, gitdir=os.path.join(worktree, '.git')) as git:
-        src_file = os.path.join(worktree, 'src/java/org/pantsbuild/Class.java')
-        with safe_open(src_file, 'w') as fp:
-          fp.write(dedent("""
-          package org.pantsbuild;
-
-          class Class {
-            static final int MEANING_OF_LIFE = 42;
+  @contextmanager
+  def known_commits_in(self, worktree, prefix=None):
+    """Creates a git repository in the given target git root directory.
+
+    If prefix is specified, it represents a tuple of a relative buildroot, and a relative source
+    dir under that buildroot.
+    """
+    gitdir = os.path.join(worktree, '.git')
+    if os.path.exists(gitdir):
+      raise Exception('`known_commits_in` should not be used with an existing git repository.')
+
+    buildroot = os.path.join(worktree, prefix[0]) if prefix else get_buildroot()
```

Comment:
This is confusing to me. I think "worktree" is the same thing as "gitroot" in the change description? If so it would be good to standardize on a single term. 

More confusingly, if I'm grokking correctly, if prefix is specified then the test buildroot is under worktree (which is under the real buildroot), otherwise the test buildroot is the real buildroot, which means it contains worktree?  

I feel like I must be reading this wrong.